crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -209,10 +230,25 @@ impl ServerBuilder {
         Ok(cors)
     }
 
+    /// Consumes the `ServerBuilder` to create a `Server` that can be run. Spawns a background task
+    /// to periodically update the `AvailableRange`, which is the range of checkpoints that the
+    /// service is guaranteed to produce a consistent result for.
     pub fn build(self) -> Result<Server, Error> {
-        let (address, schema, router) = self.build_components();
+        let metrics = self.state.metrics.clone();
+        let sleep_ms = self.state.connection.available_range_update_ms;
+        let (address, schema, db_reader, router) = self.build_components();
+
+        // Start the background task to update a shared `AvailableRange`
+        let shared_available_range: Arc<RwLock<AvailableRange>> =
+            Arc::new(RwLock::new(AvailableRange { first: 0, last: 0 }));
+        let available_range_clone = shared_available_range.clone();
+        spawn_monitored_task!(async move {
```

Comment:
Does this thread ever get cleaned up? At the moment, it seems like we are just leaking it which is not good for test environments where we may be creating many of these.

Reply:
Made some changes to support cleaning the thread up. Explains why the e2e tests took much longer